src/typegenMetadata.ts

```diff
@@ -21,40 +20,28 @@ import {
  * generated types and/or SDL artifact, including but not limited to:
  */
 export class TypegenMetadata {
-  protected typegenFile: string = "";
-
-  constructor(protected config: BuilderConfig) {
-    if (config.outputs !== false && config.shouldGenerateArtifacts !== false) {
-      if (config.outputs.typegen) {
-        this.typegenFile = assertAbsolutePath(
-          config.outputs.typegen,
-          "outputs.typegen"
-        );
-      }
-    }
-  }
+  constructor(protected config: BuilderConfig) {}
 
   /**
    * Generates the artifacts of the build based on what we
    * know about the schema and how it was defined.
    */
   async generateArtifacts(schema: NexusSchema) {
     const sortedSchema = this.sortSchema(schema);
-    if (this.config.outputs) {
-      if (this.config.outputs.schema) {
-        await this.writeFile(
-          "schema",
-          this.generateSchemaFile(sortedSchema),
-          this.config.outputs.schema
-        );
-      }
-      if (this.typegenFile) {
-        const value = await this.generateTypesFile(
-          sortedSchema,
-          schema.extensions.nexus
-        );
-        await this.writeFile("types", value, this.typegenFile);
-      }
+    if (this.config.outputs.schema) {
+      await this.writeFile(
```

Comment:
I think @Weakky mentioned wanting to change from `writeFile` to `unlink / writeFile`, apparently vscode picks up the changes better this way if the file is replaced rather than modified? Not sure if we should do that as part of this or in a separate PR.

Reply:
@tgriesser sounds great, but yeah separate PR 👍